packages/dds/tree/src/test/shared-tree/treeNodeApi.spec.ts

```diff
@@ -56,59 +56,480 @@ describe("treeApi", () => {
+			});
+
+			it("success + user defined value", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return { value: view.root.content };
+				});
+				assert.equal(view.root.content, 43);
+				assert.equal(runTransactionResult.success, true);
+				assert.equal(runTransactionResult.value, 43);
+			});
+
+			it("rollback + user defined value", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return { rollback: true, value: view.root.content };
+				});
+				assert.equal(runTransactionResult.success, false);
+				// Note that this is the value that was set before the transaction was rolled back.
+				assert.equal(runTransactionResult.value, 43);
+				// After the transaction is rolled back, the value is reverted to the original value.
+				assert.equal(view.root.content, 42);
+			});
+
+			it("success + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(view.root.content, 43);
+				assert.equal(runTransactionResult.success, true);
+			});
+
+			it("rollback + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						rollback: true,
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(view.root.content, 42);
+				assert.equal(runTransactionResult.success, false);
+			});
+
+			it("success + user defined value + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						value: view.root.content,
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(view.root.content, 43);
+				assert.equal(runTransactionResult.success, true);
+				assert.equal(runTransactionResult.value, 43);
+			});
+
+			it("rollback + user defined value + undo precondition", () => {
+				const view = getTestObjectView();
+				const runTransactionResult = view.runTransaction(() => {
+					view.root.content = 43;
+					return {
+						rollback: true,
+						value: view.root.content,
+						undoPreconditions: [{ type: "nodeInDocument", node: view.root }],
+					};
+				});
+				assert.equal(runTransactionResult.success, false);
+				// Note that this is the value that was set before the transaction was rolled back.
+				assert.equal(view.root.content, 42);
+				// After the transaction is rolled back, the value is reverted to the original value.
+				assert.equal(runTransactionResult.value, 43);
+			});
```

Comment:
I don't think we need to test every permutation of return fields. So long as each return field is tested, I think we can remove these two tests.

Reply:
removed them.